beater/beater.go

```diff
@@ -45,19 +43,17 @@ func (bt *ApmServer) Run(b *beat.Beat) error {
 		go bt.client.PublishAll(events)
 	}
 
-	bt.server, err = server.New(bt.config.Server)
-	if err != nil {
-		return err
-	}
-	bt.server.Start(callback, bt.config.Host)
-	defer bt.server.Stop()
+	server := apmServer(bt.config, callback)
+	start(server, bt.config.SSL)
```

Comment:
Why did you change this? You pass a reference to start and stop so you could directly use the object itself. It seems to me `start` and `stop` methods on the server itself are very natural. 

Also in case you use the object, you don't have to pass config.SSL as the config is already passed the constructor above.

In general I'm in favor of using objects where we can as it leads to less params passed to methods and works very well in the idea as I directly see which methods are available on an object.

But as this is also more a difference on the way the two of us code, I would like to get the input from @simitt and @roncohen on their preference as otherwise we go back and forth on this one.